packages/iris-grid/src/IrisGridProxyModel.ts

```diff
@@ -476,6 +493,74 @@ class IrisGridProxyModel extends IrisGridModel {
     this.model.filter = filter;
   }
 
+  get partition(): readonly unknown[] {
+    return this.originalModel.partition;
+  }
+
+  set partition(partition: readonly unknown[]) {
+    log.debug('set partition', partition);
+    this.originalModel.partition = partition;
+    if (isIrisGridPartitionedTableModel(this.originalModel)) {
+      if (partition.length === 0) {
+        this.setNextModel(
+          this.originalModel.partitionedTable.getMergedTable().then(table => {
+            const newModel = makeModel(this.dh, table, this.formatter);
+            (this.originalModel as IrisGridPartitionedTableModel).model =
```

Comment:
We shouldn't be using `IrisGridPartitionedTableModel` as a proxy like this - we've already got the `IrisGridProxyModel` as a proxy... and now we're mixing the responsibility between the two here.
`IrisGridPartitionedTableModel` should just be the top level partitioned table model I think, don't need to set a child model in there. Just do `setNextModel` here.